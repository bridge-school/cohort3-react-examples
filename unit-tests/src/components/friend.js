import React from "react";

export const Friend = ({ firstName, last }) => (
  <div>
    <div>{firstName}</div>
    <div>{last}</div>
  </div>
);
