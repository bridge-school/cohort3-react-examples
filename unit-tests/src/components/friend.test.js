import React from "react"; // need to import to write JSX
import renderer from "react-test-renderer"; // since we aren't mounting to the DOM, we need to mock render component trees
import { Friend } from "./friend"; // import the component to test

test("Friend component renders with a persons first and last name", () => {
  const component = renderer.create(<Friend firstName={"Jon"} last={"Snow"} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

// write a test to check that if no last name is provided, it only displays the firstname
