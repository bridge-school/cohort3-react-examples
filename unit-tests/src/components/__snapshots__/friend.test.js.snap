// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Friend component renders with a persons first and last name 1`] = `
<div>
  <div>
    Jon
  </div>
  <div>
    Snow
  </div>
</div>
`;
